node/src/components/block_synchronizer/block_acquisition_action.rs

```diff
@@ -296,7 +300,469 @@ impl BlockAcquisitionAction {
                 validator_weights,
                 signatures,
                 is_historical,
+                legacy_required_finality,
             ),
         }
     }
 }
+
+fn enough_signatures(
+    signature_weight: SignatureWeight,
+    legacy_required_finality: LegacyRequiredFinality,
+    is_checkable: bool,
+    is_historical: bool,
+) -> bool {
+    use SignatureWeight::*;
+
+    if is_checkable {
+        // Normal blocks:
+        signature_weight.is_sufficient(is_historical)
```

Comment:
I think this was a bug in the existing code, but it should be `!is_historical`

Reply:
Done